src/OpenSearch.Client/QueryDsl/Abstractions/Container/QueryContainer-Dsl.cs

```diff
@@ -131,5 +131,42 @@ public void Accept(IQueryVisitor visitor)
 
 		// ReSharper disable once UnusedMember.Global
 		internal bool ShouldSerialize(IJsonFormatterResolver formatterResolver) => IsWritable;
+
+
+		public QueryContainer Name(string name)
+		{
+			ContainedQuery.Name = name;
+			return this;
+		}
+
+		/// <summary>
+		/// Applies or removes the `strict` attribute to the query container and all child sub-queries.
+		/// </summary>
+		/// <param name="strict">When null, the value is taken from the root query.</param>
```

Comment:
This "When null, .." behaviour feels somewhat unintuitive for what a null value should do and I'm unsure of the use-case. It might be better to keep that as something that's done intentionally via something like `container.Strict(container.IsStrict, recurse: true)`

Reply:
Yes, I was initially going for a method signature of `Strict(bool isStrict)`, but then some tests requiring special method signature for single-argument methods on `IDescriptor` started failing. I am not sure why those requirements exist. But your suggestion seems nice.

Reply:
I think I know what tests you're talking about, if you just ignore that test for now, then we can figure out the best way to have the test ignore this afterwards